lib/evaluate/tools.h

```diff
@@ -103,13 +102,12 @@ template<typename A> bool IsVariable(const A &x) {
 }
 
 // Predicate: true when an expression is assumed-rank
-bool IsAssumedRank(const semantics::Symbol &);
+bool IsAssumedRank(const Symbol &);
 bool IsAssumedRank(const ActualArgument &);
 template<typename A> bool IsAssumedRank(const A &) { return false; }
 template<typename A> bool IsAssumedRank(const Designator<A> &designator) {
-  if (const auto *symbol{
-          std::get_if<const semantics::Symbol *>(&designator.u)}) {
-    return IsAssumedRank(*symbol);
+  if (const auto *symbol{std::get_if<SymbolRef>(&designator.u)}) {
+    return IsAssumedRank(symbol->get());
```

Comment:
It is probably moot now, but I don't understand how this could have worked before.

Now `symbol` is `const SymbolRef *` so `symbol->get()` is `const Symbol &`, as expected.

But before it looks like `symbol` would have been `const Symbol **` and so a `const Symbol *` would have been passed to `IsAssumedRank`, which expects `const Symbol &`.

Reply:
`IsAssumedRank()` is a template function with a default catch-all case that returns `false`.  So the original code compiled but didn't work properly.